pandas/io/parsers.py

```diff
@@ -2153,8 +2155,12 @@ def _try_convert_dates(parser, colspec, data_dict, columns):
 
     new_name = '_'.join([str(x) for x in colnames])
     to_parse = [data_dict[c] for c in colnames if c in data_dict]
-
     new_col = parser(*to_parse)
+    if hasattr(new_col, 'dtype') and any(r in str(new_col.dtype) for r in
```

Comment:
Oh, I see what's wrong. I had previously committed the two unwanted commits to my fork's master branch (before pulling the updated master branch). I tried `git revert` but that caused merge conflicts so I reset. How can we solve this with a minimum of mess? Should we just close this PR and I'll open a new one with the good commits?